src/xinterpreter.cpp

```diff
@@ -106,6 +107,8 @@ namespace xpyt
             return kernel_res;
         }
 
+        forward_input(allow_stdin);
```

Comment:
You should maybe call `forward_input` only if `allow_stdin` is true? And remove the argument for the `forward_input` method, as you don't use it.

Reply:
Resolved as implemented in a scope guard.